flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/JobMasterServiceLeadershipRunnerTest.java

```diff
@@ -662,15 +663,21 @@ public void testJobMasterServiceProcessCreationFailureIsForwardedToResultFuture(
 
     @Test
     public void testJobAlreadyDone() throws Exception {
-        JobID jobID = new JobID();
+        JobID jobId = new JobID();
+        JobResult jobResult =
+                new JobResult.Builder()
+                        .applicationStatus(ApplicationStatus.UNKNOWN)
+                        .jobId(jobId)
+                        .netRuntime(Long.MAX_VALUE)
+                        .build();
         try (JobManagerRunner jobManagerRunner =
                 newJobMasterServiceLeadershipRunnerBuilder()
                         .setJobMasterServiceProcessFactory(
                                 TestingJobMasterServiceProcessFactory.newBuilder()
-                                        .setJobId(jobID)
+                                        .setJobId(jobId)
                                         .build())
                         .build()) {
-            runningJobsRegistry.setJobFinished(jobID);
+            jobResultStore.createDirtyResult(jobResult);
```

Comment:
nit: I would move this line out of the try/catch block to group the test initialization in a single code block